WordPress/gutenberg
# Block Editor: Refactor `LinkControl` tests to `@testing-library`

## What?
This PR refactors the `<LinkControl />` tests to use `@testing-library` instead of `react-dom` and its testing utils.

## Why?
We've been aiming to have all our component tests resemble user behavior as much as possible, so this is a step in that direction.

## How?
This is a complex PR with many different types of changes, namely:

* Using `@testing-library/react`'s `render()` instead of `react-dom`'s render.
* Using `@testing-library/user-event` instead of `react-dom/test-utils` `Simulate`.
* Using `@testing-library/react`'s `act()` instead of the `react-dom/test-utils` one.
* Removing `act()` where it is redundant.
* Using `screen` queries instead of `querySelector` and `querySelectorAll()`
* Improving some assertions - like `toBeTruthy()` -> `toBeVisible()`.
* Reducing the DOM traversing - less `.innerHTML`, and more `jest-dom` assertions instead.

There's a bug with the `keyDown` events - they simply don't work with `@testing-library/user-event`, as for some reason the events dispatched there have a `keyCode: 0`. That's why we're falling back to `fireEvent`, where we can actually provide the `keyCode` ourselves and get around that problem.

There are a few more TODOs (and unadressed items from the ones listed above) in the file we're touching, but some of them will likely need changes to the components themselves, and this PR has grown a bit more than I expected, so I'm happy to follow-up with addressing those in another PRs.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/block-editor/src/components/link-control/test/index.js`
